numba/__init__.py

```diff
@@ -104,10 +104,38 @@ def _ensure_pynumpy():
     if np_version < (1, 7):
         raise ImportError("Numba needs Numpy 1.7 or greater")
 
+def _try_enable_svml():
+    """
+    Tries to enable SVML if configuration permits use and the library is found.
+    """
+    if not config.DISABLE_INTEL_SVML:
+        try:
+            if sys.platform.startswith('linux'):
+                llvmlite.binding.load_library_permanently("libsvml.so")
+            elif sys.platform.startswith('darwin'):
+                llvmlite.binding.load_library_permanently("libsvml.dylib")
+            elif sys.platform.startswith('win'):
+                llvmlite.binding.load_library_permanently("svml_dispmd")
+            else:
+                return False
+            llvmlite.binding.set_option('SVML', '-vector-library=SVML')
+            return True
+        except:
+            if platform.machine == 'x86_64' and config.DEBUG:
+                warnings.warn("SVML was not found/could not be loaded.")
+    return False
 
 _ensure_llvm()
 _ensure_pynumpy()
 
+# we know llvmlite is working as the above tests passed, import it now as SVML
+# needs to mutate runtime options (sets the `-vector-library`).
+import llvmlite
+
+"""
+Is set to True if Intel SVML is in use.
+"""
+config.USING_SVML = _try_enable_svml()
```

Comment:
This is what I was worried about/trying to avoid with https://github.com/anton-malakhov/numba/pull/1#discussion_r178973296, which I didn't really like either, essentially preserving state in the configuration system or `__init__`. @sklam do you have a view on this?

Reply:
@stuartarchibald, I understand your worry.  I think we can defer it to the creation of the cpu target-context at https://github.com/numba/numba/blob/35d29779650ef89c8a7b23dd006e73663055cad5/numba/targets/cpu.py#L40-L49

Reply:
there might be multiple CPUContexts while svml must be loaded just once, right? So, we still need a global variable indicating whether SVML was loaded.  Whether LLVM was initialized with the SVML option might be separated if LLVM can be reconfigured in run-time

Reply:
nevermind, it can't go into the cpu target-context because it may initialize more than.  So, I can't think of a better place to put it.